consai_examples/consai_examples/decisions/goalie.py

```diff
@@ -55,8 +55,10 @@ def inplay(self, robot_id):
         ID_IN_DEFENSE = self.ACT_ID_INPLAY + 0
         ID_IN_PLAY = self.ACT_ID_INPLAY + 1
 
+        if self._field_observer.ball_is_moving():
+            self._defend_goal(robot_id)
         # ボールがディフェンスエリアにあるときは、ボールを蹴る
-        if self._ball_state == FieldObserver.BALL_IS_IN_OUR_DEFENSE_AREA:
+        elif self._ball_state == FieldObserver.BALL_IS_IN_OUR_DEFENSE_AREA:
```

Comment:
- このif文にelseが存在していない
- 58行目でself._act_idを更新していない

多分上記の理由で、ディフェンスエリア外でボールが止まっているときに行動が定まらずにゴーリーが暴走しました。  
(ID_IN_PLAYのままなので下の処理 `_defend_goal_with_kick` にも入れない)

```suggestion
        if not self._field_observer.ball_is_moving() and self._ball_state == FieldObserver.BALL_IS_IN_OUR_DEFENSE_AREA:
```

58-59行目のifを消して上記のように変更して ID_IN_PLAYの動きを `_defend_goal` にしてしまえば良い気がします

Reply:
上記の作りにして、 `if self._act_id != ID_IN_PLAY: ...` の処理は前のまま（if文の外に出す）が良いかなと思う